addons/hr_expense/wizard/hr_expense_split_wizard.py

```diff
@@ -52,6 +52,14 @@ def action_split_expense(self):
                     for attachment in attachment_ids:
                         attachment.copy({'res_model': 'hr.expense', 'res_id': coplied_expense.id})
 
+            return {
+                'type': 'ir.actions.act_window',
+                'name': _('My Expenses'),
+                'view_mode': 'tree,form',
+                'res_model': 'hr.expense',
+                'view_id': False,
+            }
+
```

Comment:
You don't need to do that, 42c1d2a6 broke that by adding a views key in the action dict that conflicts with the view_mode (for unkown reasons but probably a mistake)
Just reverting that line should do (for the wrong redirection)